docs/DESIGN.md

```diff
@@ -1,95 +1,187 @@
 
-- DSCInitialization
-- DataScienceCluster
+In addition, each ODH component is represented by its own dedicated internal API/custom resource.
+Each component CR singleton instance and its sub-resources are managed by a dedicated component controller.
+More details on the supported ODH components are provided [here](#component-crs-and-controllers).
 
-![Operator Re-architecture](images/Operator%20Architecture.png)
+The operator also watches [Feature Trackers](#feature-trackers) which are cluster-scoped resources, specifically designed for monitoring and managing objects created via the internal Features API.
 
-### DSCInitialization
+Furthermore, there are dedicated [service CRs and controllers](#service-crs-and-controllers) for configuring authentication and monitoring.
 
-- This CR will be created by the ODH operator to perform initial setup that is common for all components
-- Some examples of initial setup include creating namespaces, network policies, SCCs, common configmaps and secrets.
-- This will be a singleton CR i.e 1 instance of this CR will always be present in the cluster.
-- DSCInitialization CR can be deleted to re-run initial setup without requiring re-build of the operator.
+Lastly, the ODH operator also manages [accessory controllers](#accessory-controllers) responsible for helper tasks, such as secret generation and cleanup after uninstallation.
+
+A high-level diagram showcasing the majority of the above-mentioned CRs and their reconcilers is provided below.
+
+![Operator Reconciler Architecture](images/odh-operator-reconciliation.png)
 
-### DataScienceCluster
+### User-facing Custom Resources and their controllers
 
-- This CR will be watched by the ODH operator to enable various data science components.
-- It is responsible for enabling support for CRDs like Notebooks, DataSciencePipelinesApplication, InferenceService etc. based on the configuration
-- Initially only one instance of DataScienceCluster CR will be supported by the operator. A user can extend/update the CR to enable/disable components.
-- Detailed API fields are described in the CRD.
+The two user-facing CRs are **DSCInitialization** (DSCI) and **DataScienceCluster** (DSC).
+Together, these CRs are responsible for configuration of ODH at the platform level.
+
+#### DSCInitialization (DSCI)
+
+- DSCI CR is created and managed by the ODH operator to perform initial setup that is common for all components.
+  - for example, DSCI defines the namespace for ODH component deployments, monitoring config, ServiceMesh config,...
+- Some examples of initial setup include creating namespaces, network policies, SCCs, common configmaps and secrets.
+- DSCI instance is a singleton in the k8s/OCP cluster, i.e., only one instance of this CR will always be present in the cluster.
+- DSCI can be deleted to re-run the initial setup without requiring rebuild of the operator.
```

Comment:
does this need to be here?

Reply:
I pretty much kept (and slightly updated) these DSCI and DSC descriptions from the previous version of the document. My thinking was to keep it, as these are the main user-facing CRs and it helped me to understand their purpose

Reply:
I see, I'll remove it, thanks